### PR TITLE
Add HTTP custom method example to doc

### PR DESCRIPTION
**Pull request by richard-imaoka**
Refs #954 

Adding description and a code example for HTTP custom methods in doc.

### CODE REVIEW

File: docs/src/main/paradox/java/http/common/http-model.md [hunk 1]
```diff
@@ -346,6 +346,14 @@ Similarily to the media types registration, you can register custom status codes
 
 @@snip [CustomStatusCodesExampleTest.java](../../../../../test/java/docs/http/javadsl/CustomStatusCodesExampleTest.java) { #application-custom-java }
 
+<a id="registeringcustommethod-java"></a>
+## Registering Custom HTTP Method
+
+Akka HTTP also allows you to define custome HTTP methods, other than the well-known methods @javadoc[predefined](akka.http.javadsl.model.HttpMethods) in Akka HTTP.
+To use a custom HTTP method, you need to define it, and then add it to parser settings like below:
```
Comment (jlprat) [inline docs/src/main/paradox/java/http/common/http-model.md]: Nitpicking here, but maybe say "define it first, and then..."
`To use a custom HTTP method, you need to define it first, and then add it to parser settings like below:`

File: docs/src/test/scala/docs/http/scaladsl/server/directives/CustomHttpMethodSpec.scala [hunk 2]
```diff
@@ -30,7 +30,7 @@ class CustomHttpMethodSpec extends AkkaSpec with ScalaFutures
       //#application-custom
       import akka.http.scaladsl.settings.{ ParserSettings, ServerSettings }
 
-      // define custom media type:
+      // define custom method type:
       val BOLT = HttpMethod.custom("BOLT", safe = false,
         idempotent = true, requestEntityAcceptance = Expected)
```
Comment (jlprat) [inline docs/src/test/scala/docs/http/scaladsl/server/directives/CustomHttpMethodSpec.scala]: Move the ending `//#application-custom` 4 lines below (after `val response = Http().singleRequest(request).futureValue`), so the request example is included.

File: docs/src/test/scala/docs/http/scaladsl/server/directives/CustomHttpMethodSpec.scala [hunk 3]
```diff
@@ -30,7 +30,7 @@ class CustomHttpMethodSpec extends AkkaSpec with ScalaFutures
       //#application-custom
       import akka.http.scaladsl.settings.{ ParserSettings, ServerSettings }
 
-      // define custom media type:
+      // define custom method type:
```
Comment (jlprat) [inline docs/src/test/scala/docs/http/scaladsl/server/directives/CustomHttpMethodSpec.scala]: nice catch

File: docs/src/main/paradox/java/http/common/http-model.md [hunk 4]
```diff
@@ -346,6 +346,14 @@ Similarily to the media types registration, you can register custom status codes
 
 @@snip [CustomStatusCodesExampleTest.java](../../../../../test/java/docs/http/javadsl/CustomStatusCodesExampleTest.java) { #application-custom-java }
 
+<a id="registeringcustommethod-java"></a>
```
Comment (ktoso) [inline docs/src/main/paradox/java/http/common/http-model.md]: I actually don't know if we need to put these manually, do we @2m @johanandren ?

File: docs/src/test/scala/docs/http/scaladsl/server/directives/CustomHttpMethodSpec.scala (hunk 3 above)
Comment (ktoso) [inline docs/src/test/scala/docs/http/scaladsl/server/directives/CustomHttpMethodSpec.scala]: yup :)

File: docs/src/test/scala/docs/http/scaladsl/server/directives/CustomHttpMethodSpec.scala (hunk 2 above)
Comment (ktoso) [inline docs/src/test/scala/docs/http/scaladsl/server/directives/CustomHttpMethodSpec.scala]: Done this

**Comment by ktoso** [comment]
I amended some small things, LGTM - thanks !

**Pull request merged**